datafusion/sql/src/planner.rs

```diff
@@ -502,7 +502,14 @@ pub fn object_name_to_table_reference(
     let ObjectName(idents) = object_name;
     idents_to_table_reference(idents, enable_normalization)
 }
-
+fn substitute_tilde(cur: String) -> String {
+    if let Ok(usr_dir) = env::var("HOME") {
```

Comment:
Since datafusion-cli already uses `dirs` https://github.com/apache/arrow-datafusion/blob/e62240969135e2236d100c8c0c01546a87950a80/datafusion-cli/Cargo.toml#L48

What about using https://docs.rs/dirs/latest/dirs/fn.home_dir.html ?